showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    httpJsonClientWithRetrySetting.close();
+    httpJsonClientWithRetrySetting.awaitTermination(
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_attemptPermanentFailure_recordsMetrics() throws Exception {
+
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory10",
+        "../opentelemetry-helper/configs/testGrpc_attemptPermanentFailure.yaml");
+
+    RetrySettings retrySettings = RetrySettings.newBuilder().setMaxAttempts(3).build();
+
+    EchoStubSettings.Builder grpcEchoSettingsBuilder = EchoStubSettings.newBuilder();
+    grpcEchoSettingsBuilder
+        .blockSettings()
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(ImmutableSet.of(Code.NOT_FOUND));
+
+    EchoSettings grpcEchoSettings = EchoSettings.create(grpcEchoSettingsBuilder.build());
+    grpcEchoSettings =
+        grpcEchoSettings
+            .toBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4326"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .setEndpoint("localhost:7469")
+            .build();
+
+    EchoClient grpcClientWithRetrySetting = EchoClient.create(grpcEchoSettings);
+
+    BlockRequest blockRequest =
+        BlockRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.INVALID_ARGUMENT.ordinal()).build())
+            .build();
+
+    grpcClientWithRetrySetting.blockCallable().futureCall(blockRequest).isDone();
+
+    Thread.sleep(5000);
+
+    String filePath = "../directory10/testGrpc_attemptPermanentFailure_metrics.txt";
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    grpcClientWithRetrySetting.close();
+    grpcClientWithRetrySetting.awaitTermination(
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  // Helper function for creating Opentelemetry object with a different port for exporter for every
+  // test
+  // this ensures that logs for each test are collected separately
+  private static ApiTracerFactory createOpenTelemetryTracerFactory(String port) {
+    // OTLP Metric Exporter setup
+    String endpoint = "http://localhost:" + port;
+    OtlpGrpcMetricExporter metricExporter =
+        OtlpGrpcMetricExporter.builder().setEndpoint(endpoint).build();
+
+    // Periodic Metric Reader configuration
+    PeriodicMetricReader metricReader =
+        PeriodicMetricReader.builder(metricExporter)
+            .setInterval(java.time.Duration.ofSeconds(3))
```

Comment:
qq, Is there a reason specifically for 3s as the interval? Perhaps that's what seems like a good default value or something else?

Reply:
No specific reason, yes, that is like a good default value I could think of, given every test waits for 5 seconds for the metrics to get exported.

Reply:
Ok, I think that's fine. Can you document that in the comments or create a static constant for the 3s value?